rails dbconsole defaults to include_password

wpeterson opened the pull request:
### Summary

`rails dbconsole` should default to `include_password`

rails dbconsole will inject the password into the DB console if one is configured and the `include_password` flag is enabled.  Let’s enable this flag by default.

For years, developers have been running `rails dbconsole` in staging or production and pasting in a password, not even knowing that `rails dbconsole -p` would do what they actually wanted and apply the credentials already configured in their `config/database.yml`.  Let's apply the principle of least surprise and change the default behavior to what most folks would expect and improve the user experience.

### Other Information

* fixes https://github.com/rails/rails/issues/45901

wpeterson committed:
rails dbconsole should default to include_password

rails dbconsole will inject the password into the DB console if one is configured and the include_password flag is enabled.  Let’s enable this flag by default.

wpeterson committed:
Update tests for dbconsole default include_password

skipkayhil commented on railties/lib/rails/commands/dbconsole/dbconsole_command.rb:
```diff
@@ -167,7 +167,7 @@ module Command
       include EnvironmentArgument
 
-      class_option :include_password, aliases: "-p", type: :boolean,
+      class_option :include_password, aliases: "-p", type: :boolean, default: true,
         desc: "Automatically provide the password from database.yml"
```
If we're setting the default value of a boolean param to true, would it even be possible for it to ever be false? Seems like the param should be deprecated/removed if it isn't going to have an effect any more.

This will also definitely need a changelog entry

wpeterson replied:
Thanks for looking at this PR! I really appreciate it.

The dbconsole command documents flags to both enable/disable this option:

```
Usage:
  rails dbconsole [options]

Options:
  -e, [--environment=ENVIRONMENT]                    # Specifies the environment to run this dbconsole under (test/development/production).
  -p, [--include-password], [--no-include-password]  # Automatically provide the password from database.yml
                                                     # Default: true
      [--mode=MODE]                                  # Automatically put the sqlite3 database in the specified mode (html, list, line, column).
                                                     # Possible values: html, list, line, column
      [--header], [--no-header]
  --db, [--database=DATABASE]                        # Specifies the database to use.
  ```

I believe boolean options provide a command line flag to both enable or disable the value.  Currently, this option is disabled, but running `bin/rails dbconsole --include-password` will enable it.

With this PR changing the default value to true, it can still be disabled if users specify the `bin/rails dbconsole --no-include-password` flag on the command.  I've just confirmed this flag will still disable the value even with a default of true.

wpeterson replied:
I've added a `railties` CHANGELOG entry in b98e8918749b424f95552dbb6f11a6d481609636 describing this behavior change.

wpeterson committed:
Add rails dbconsole include_password CHANGELOG

Describe the new default value/behavior for the dbconsole credential handling via `include_password` option value

rafaelfranca commented:
Thank you for the pull request. The reason why this isn't default is that the password goes to the history logging of commands, and on the PID of the process which will very much cause security problems.

See https://github.com/rails/rails/commit/0abf0da0016abc455145810d7060a10e0b56b0b6

akostadinov commented:
Both `mysql` and `psql` support providing password through an options file. It is more correct to avoid the security concerns that way, instead of leaving responsibility to clueless humans to obtain and supply password to the remote host securely.

zzak commented:
@akostadinov I'm not sure but a quick google search says at least for postgresql to use the `--passfile` flag for that, is that what you mean? [[ref](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-PASSFILE)]

akostadinov commented:
@zzak , I didn't find this option but it should do the job. For MySQL I think there is no such flag. But one can use a custom options file (`my.cnf`).
